bundles/org.smarthomej.binding.telenot/src/main/java/org/smarthomej/binding/telenot/internal/handler/IPBridgeHandler.java

```diff
@@ -181,8 +181,11 @@ protected synchronized void disconnect() {
         if (s != null) {
             try {
                 s.close();
+                Thread.sleep(2000);
```

Comment:
Why do you add a sleep here? In general sleep should be avoided.

Reply:
I want to wait 2 seconds before the reconnect. Can you help for a better way?

Reply:
I guess this issue is only relevant for the call to `disconnect` and the beginning of `connect`, is that right? Then you could refactor connect to

```
public void connect() {
    disconnect();
    ScheduledFuture<> connectJob = this.connectJob;
    if (connectJob == null || connectJob.isDone()) { 
        connectJob = scheduler.schedule(this::internalConnect, 2, TimeUnit.SECONDS);
    }
}
```

and move the rest of the code to `internalConnect()`. Make sure that `connectJob` is also cancelled on disposal.